docs/src/modules/java/pages/actions-as-controller.adoc

```diff
@@ -3,86 +3,177 @@
-an Action that will take the incoming `Increase` command, double its value and forward the request to a `Counter` entity.
+== Forwarding Commands
 
-Note that in this example we are adding an Action to an existing service, the xref:value-entity.adoc[Value Entity Counter] service.
+The `forward` effect allows us to transform or further validate an incoming request before passing it on to another
+component and have the response message directly passed back to the client making the request. The response from the
+forwarded operation must have the same response type as the original command.
+
+In this example we accept the same command as the entity, `AddLineItem`, but add some additional verification of the
+request and only conditionally forward the request to the entity if the verification is successful:
 
 [.tabset]
 Java::
 +
-[source,proto,indent=0]
-.src/main/proto/com/example/actions/double-counter.proto
+.src/main/java/com/example/shoppingcart/ShoppingCartActionImpl.java
+[source,java]
 ----
-include::example$java-valueentity-counter/src/main/proto/com/example/actions/double-counter.proto[tag=double-action-proto]
+include::example$java-valueentity-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartActionImpl.java[tag=forward]
 ----
-<1> Import the Counter API definition
-<2> The protobuf option (akkaserverless.service) is specific to code-generation as provided by the Akka Serverless Maven plugin. This annotation indicates to the code-generation that an Action must be generated.
-<3> The Action mimics the Counter API
+<1> check if the added item is carrots
+<2> if it is "carrots" immediately return an error, disallowing adding the item
+<3> for allowed requests, use `components().shoppingCart().addItem()` to get a `DeferredCall`
+<4> The `DeferredCall` is used with `effects().forward()` to forward the request to the entity
 
 Scala::
 +
-[source,proto,indent=0]
-.src/main/proto/com/example/actions/double-counter.proto
+.src/main/java/com/example/shoppingcart/ShoppingCartActionImpl.scala
+[source,scala]
 ----
-include::example$scala-valueentity-counter/src/main/proto/com/example/actions/double-counter.proto[tag=double-action-proto]
+include::example$scala-valueentity-shopping-cart/src/main/scala/com/example/shoppingcart/ShoppingCartActionImpl.scala[tag=forward]
 ----
-<1> Import the Counter API definition
-<2> The protobuf option (akkaserverless.service) is specific to code-generation as provided by the Akka Serverless sbt plugin. This annotation indicates to the code-generation that an Action must be generated.
-<3> The Action mimics the Counter API
+<1> check if the added item is carrots
+<2> if it is "carrots" immediately return an error, disallowing adding the item
+<3> for allowed requests, use `components.shoppingCart.addItem()` to get a `DeferredCall`
+<4> The `DeferredCall` is used with `effects.forward()` to forward the request to the entity
+
+
+== Async Replies (FIXME better title)
```

Comment:
Transform response from DeferredCall

?

Reply:
Went with "Transform Request and Response to Another Component"